README.md

```diff
@@ -3,13 +3,37 @@
 - List 3 reasons why asymptotic analysis may be misleading with respect to
   actual performance in practice.
 
+  - asymptotic analysis is used to compare growth rates for algorithms. This means that constant factors and lower order factors are ignored. a function that has a practical runtime of 2n^2 + 5 would have a much better performance on smaller data sets than a fuction with a runtime of 4242n + 7log(n) + 5555, even if their asymptotic complexities would be evaluated to O(n^2) and O(n) respectively.
+ 
+  - asymptotic analysis operates under assumptions about the data set used, ie best case, worst case, average case. In reality data used may adhere to certain patterns, such as always being partially or nearly sorted. algorithms that have good best case but bad average or worst case complexity may be faster on said data sets, while an algorithm with the same best worst and average case may be less effective.
+ 
+  - hardware and environmental factors can have an affect on the time taken and can affect different algorithms and different implementations different. I am currently in computer organization while taking this class so it's fun to see these two courses intersect like this. The architecture of a system can work to the advantage or disadvantage of an algorithm. For example, non-contiguous storage access may take longer or result in more misses than something that uses contiguous storage, a factor not considered by asymptotic complexity.
+
 - Suppose finding a particular element in a binary search tree with 1,000
   elements takes 5 seconds. Given what you know about the asymptotic complexity
   of search in a binary search tree, how long would you guess finding the same
   element in a search tree with 10,000 elements takes? Explain your reasoning.
 
+  a binary search has a time complexity of log2(n)
+  plug in 5 seconds and 1000 elements to determine a theoretical constant factor
+  5 = (k)log2(1000)
+  5 = (k)9.97
+  k = .5
+  plug back in 10000
+  x = (.5)log2(10000)
+  x = (.5)13.29
+  x = 6.65 seconds
+
 - You measure the time with 10,000 elements and it takes 100 seconds! List 3
   reasons why this could be the case, given that reasoning with the asymptotic
   complexity suggests a different time.
 
+  - The obvious reason that comes to mind is that perhaps the search tree with 10k elements is extremely unbalanced, leading to a structure more akin to a list than a binary search tree. This can cause search time to go from logarithmic to linear, a significantly worse case.
```

Comment:
It would depend on the level to which the tree is unbalanced (tree height) and the depth of the node we're searching for within that tree. A completely unbalanced tree, which is in essence a linked list, would have a linear runtime for search, with the runtime being equal to the exact height of the node we're searching for.

Reply:
...and then the expected runtime would be 50s, which is still much less.

Reply:
That would be assuming that tree balance was the only factor, some combination of the three reasons given could be the reason for the unexpected slowdown

Reply:
Looking for individual reasons here.

Reply:
Would the 1000 element tree being balanced and the 10000 element tree being unbalanced not be able to account for the time difference?